### PR TITLE
Apply formatting by rustfmt-preview [ECR-779]

### PR DESCRIPTION
**Pull request by DarkEld3r**
[Previously](https://github.com/exonum/exonum/pull/558) I was against applying "preview" formatting so soon. :upside_down_face: Now I highly like that idea because of "[nested import groups](https://github.com/rust-lang/rust/pull/47948)" feature. So here is the plan:
- Apply new formatting.
- Update imports.

I will _gladly_ do the second step too. Additionally, the order of imports is sometimes wrong, so I'm going to fix it according to our [conventions](https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions).

### CODE REVIEW

**Comment by DarkEld3r** [comment]
```diff
+1,361
-1,460
```
:tada:

**Comment by DarkEld3r** [comment]
@alekseysidorov proposed to specify the Rust version in the [file](https://github.com/rust-lang-nursery/rustup.rs/pull/1172).

**Comment by andreyluchin** [comment]
Looks like our `lints` build job in travis went down to 2 mins from the usual ~10 min!

File: services/configuration/src/tests/mod.rs [hunk 1]
```diff
@@ -380,9 +379,9 @@ fn test_discard_votes_with_expired_actual_from() {
     testkit.create_blocks_until(Height(10));
     let illegal_vote = new_tx_config_vote(&testkit.network().validators()[0], new_cfg.hash());
     testkit.create_block_with_transactions(txvec![illegal_vote.clone()]);
-    assert!(!testkit.votes_for_propose(new_cfg.hash()).contains(&Some(
-        illegal_vote,
-    )));
+    assert!(!testkit
+        .votes_for_propose(new_cfg.hash())
+        .contains(&Some(illegal_vote,)));
```
Comment (alexander-irbis) [inline services/configuration/src/tests/mod.rs]: This comma looks superfluous.

File: exonum/src/encoding/serialize/json.rs [hunk 2]
```diff
@@ -159,7 +159,7 @@ impl_deserialize_int!{u8; u16; u32; i8; i16; i32}
 impl_deserialize_bigint!{u64; i64}
 impl_deserialize_hex_segment!{Hash; PublicKey; Signature}
 impl_default_deserialize_owned!{u8; u16; u32; i8; i16; i32; u64; i64;
-                                Hash; PublicKey; Signature; bool}
+Hash; PublicKey; Signature; bool}
```
[the remark this answers is not in the document]
Comment (vitvakatu) [reply exonum/src/encoding/serialize/json.rs]: Let's split it manually into two (or more) macro invocations.

File: exonum/src/storage/indexes_metadata.rs [hunk 3]
```diff
@@ -209,7 +206,7 @@ mod tests {
 
     #[test]
     #[should_panic(expected = "Attempt to access index 'test_index' of type Map, \
-    while said index was initially created with type ProofMap")]
+                               while said index was initially created with type ProofMap")]
```
[the remark this answers is not in the document]
Comment (vitvakatu) [reply exonum/src/storage/indexes_metadata.rs]: Will that produce tabs in the middle of the line?

File: exonum/src/blockchain/schema.rs [hunk 4]
```diff
@@ -513,6 +514,10 @@ impl<'a> Schema<&'a mut Fork> {
         let contains = self.transactions_pool_mut().contains(hash);
         self.transactions_pool_mut().remove(hash);
         self.transactions_mut().remove(hash);
-        if contains { Ok(()) } else { Err(()) }
+        if contains {
+            Ok(())
+        } else {
+            Err(())
+        }
```
[the remark this answers is not in the document]
Comment (vitvakatu) [reply exonum/src/blockchain/schema.rs]: I'm okay with the new syntax, we're not trying to minimize lines, right?

File: exonum/src/storage/indexes_metadata.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (DarkEld3r) [reply exonum/src/storage/indexes_metadata.rs]: The test isn't broken, so I suppose not.

File: exonum/src/api/mod.rs [hunk 5]
```diff
@@ -173,10 +167,7 @@ where
 
 impl<'de, T> Deserialize<'de> for HexField<T>
 where
-    T: AsRef<[u8]>
-        + FromHex<Error = FromHexError>
-        + ToHex
-        + Clone,
+    T: AsRef<[u8]> + FromHex<Error = FromHexError> + ToHex + Clone,
```
[the remark this answers is not in the document]
Comment (DarkEld3r) [reply exonum/src/api/mod.rs]: I'm not sure: generally I would expect different type-parameters in separate lines, so this seems a little bit unnatural to me. 🙃

[the remark this answers is not in the document]
Comment (alexander-irbis) [reply exonum/src/api/mod.rs]: Here is simple enough type, so I agree that the choice to split on the line or not is not obvious enough and I would not insist. However, in the case of complex types, such as those used with tokio, the repetition of `T` in several lines immediately starts to look very natural ))

File: exonum/src/blockchain/schema.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexander-irbis) [reply exonum/src/blockchain/schema.rs]: I agree that our goal, 
    first of all, 
should be 
    the readability 
of the code.

**Pull request merged**